### DISCUSSION TITLE
N3base

### DISCUSSION THREAD

**Discussion by UTengine** [ko4life-net/ko#199] [Ideas]
I think we could seperate the game tools and server from launch options (args). It does require a lot of work,  but I think it would be neater than the whole pre processor definitions. Like we can have a parent window and child Windows we could just start a splash screen always and then create the window according to launch arguments. Knight online.exe -Tools etc what do you think @ko4life-psko  @srmeier ? This way it will be all in one solution or even project structure.

**Comment by srmeier** [comment]
What would the benefit be?

**Reply by UTengine** [reply]
> What would the benefit be?

For now not much but further down the project I do see benefits and everyone can learn from it. For the upcoming N3FX Editor/Converter it would look like Gilads Knight online engine. But more basic and not that refined yet.
https://youtu.be/UGLJHiLUWI0

**Comment by stevewgr** [comment]
I think what he is trying to say is described in more details in this ticket: https://github.com/ko4life-net/ko/issues/173

This is an architectural change that needs to be thoroughly considered before proceeding with such implementation.

**Reply by UTengine** [reply]
Exactly I was referring to that one. A well designed engine. To run tools game server simulation etc

**Comment by UTengine** [comment]
I've been following a few tutorials here and there, and I've found some bases we can work on. I would like to start of with an FX converter/editor. Since it's in a whole diff engine I would like to request for help integrating loading N3shapes and dxt's into it. At the moment I'm not interested in a whole full fledged editor (babysteps first) but just to preview some aspects and write a converter so we can convert the fxb file to the N3FXE if you guys are up for that I would be really happy. It's based on dx11 for viewmatrix and positioning with IMGUI. I've successfully converted FXB files before through fprintf console application before. If you are up for it please let me know.

**Comment by UTengine** [comment]
Closed because it seems other people don't want to help out or don't have the time for it.
